### PR TITLE
Split gdscript intro into two pages

### PR DESCRIPTION
**Pull request by balloonpopper**
Split of the gdscript basics page into two as discussed with @Calinou 

*Bugsquad edit: See discussion in https://github.com/godotengine/godot-docs/issues/2735.*

### CODE REVIEW

File: getting_started/scripting/gdscript/gdscript_basics.rst [hunk 1]
```diff
@@ -560,9 +571,13 @@ assign to it::
 Data
 ----
 
+.. _doc_gdscript_variables:
+
 Variables
 ~~~~~~~~~
 
+A variable is some data that is stored with an associated label. For example
+you could store a variable "fruit" which contained the contents "apple".
```
Comment (Calinou) [inline getting_started/scripting/gdscript/gdscript_basics.rst:580]:
```suggestion
A variable is some data that is stored with an associated label. For example
you could store a variable called ``fruit`` which contains the string ``"apple"``.
```

**Review by Calinou — APPROVED** [review]
Looks good to me. The split seems reasonable to me. It's pretty much what I had in mind when I was talking about splitting the page on IRC :slightly_smiling_face: 

I'll let others review it as well.

**Comment by balloonpopper** [comment]
It still has a higher level of assumed knowledge than I'd like it to have, but I thought it wiser to make iterative changes than rewrite the whole thing.

**Comment by balloonpopper** [comment]
@Calinou I made a small change since your review. I realised I'd left a comment in there that the keyword "pass" needed some documentation but hadn't replaced it with content before I made the commit. I've since added it, and fleshed out the text around what a function does while I was at it.

File: getting_started/scripting/gdscript/gdscript_basics.rst [hunk 2]
```diff
@@ -327,7 +327,7 @@ below for you to useful.
 
 Advanced note : Built-in types are stack-allocated. They are passed as values.
```
Comment (Calinou) [inline getting_started/scripting/gdscript/gdscript_basics.rst:328]:
```suggestion
*Advanced note:* Built-in types are stack-allocated. They are passed as values. 
```

**Comment by NathanLovato** [comment]
I would split them way more. In particular, I don't like splits into part 1 / part 2 because the titles don't help you find information faster than having it all in one page. 

Instead, I would recommend to create a new sub-section covering GDScript basics where each page covers a specific and clear topic. I'd have to go through all pages to come up with a suggestion for the exact split, but for instance variables, functions, classes (including properties and inheritance), signals, coroutines could already be 5 pages with explicit titles.

And from the original discussion, regarding the user who talked about being able to search one big page: I think for them, a GDScript cheatsheet would likely be better. Many users aren't even aware they can jump around web pages with text search or would jump around when learning the language's fundamentals.

**Comment by NathanLovato** [comment]
Note that we should probably merge #3623 and other PRs for GDScript docs first, even if they're going to change later, to avoid causing many merge conflicts.

**Comment by balloonpopper** [comment]
@NathanLovato I think it's critical for the documentation that searching explicitly for a command brings its results back as the first search.
e.g. if I search for "if", currently the link to "if/then/else" is the 10th result returned whereas it should be number 1. Searching for "var" doesn't even return the basics page at all.  
Having the search function broken, and a lack of examples for each command are the two biggest drawbacks of the documentation I believe. The structure of the documentation is of course important, but I would place it a very distant 3rd compared to a useful search and examples.

It would appear that rather than the suggested GDScript cheat sheet that search ranking could be corrected? https://docs.readthedocs.io/en/stable/config-file/v2.html#search-ranking

**Comment by NathanLovato** [comment]
The search feature isn't relevant to this issue and PR, which is about splitting a page that's too long to process well into multiple.

The docs search is a different task and a job that involves coding on the website's back-end. The person who'd likely know most about that is @Calinou.

Note that page titles seem to have a strong weight in search results and that having pages named GDScript variables or Functions in GDScript would likely be among the top results when searching respectively variable(s) or function(s). You could test that locally by making pages with only a title, building, and doing some test searches.

Also, you won't get good result in about any search engine by searching for `if` or `var`. It's too short for a keyword, too common for the engine to find something relevant.

**Comment by balloonpopper** [comment]
Closing this in light of the work that @NathanLovato is doing on the documentation.

**Comment by NathanLovato** [comment]
Ah thanks, and sorry for that!

**Pull request closed without merging**